silk-react-components/src/HierarchicalMapping/Components/MappingRule/MappingRule.jsx

```diff
@@ -16,131 +16,132 @@ import {
         // provided,
         // snapshot,
-    },
+    };
 
     // initilize state
-    getInitialState() {
+    constructor(props) {
+        super(props);
         const pastedId = sessionStorage.getItem('pastedId');
         const isPasted = (pastedId !== null) && (pastedId === this.props.id);
-        if (isPasted)
-            !sessionStorage.removeItem('pastedId');
-        return {
-            isPasted : isPasted,
+        if (isPasted) { !sessionStorage.removeItem('pastedId'); }
+        this.state = {
+            isPasted,
             expanded: isPasted || false,
             editing: false,
-            askForDiscard: false,
             loading: false,
         };
-    },
+    }
+    
     componentDidMount() {
         // listen for event to expand / collapse mapping rule
-        this.subscribe(
-            hierarchicalMappingChannel.subject('rulesView.toggle'),
-            ({expanded, id}) => {
-                // only trigger state / render change if necessary
-                if (
-                    expanded !== this.state.expanded &&
-                    this.props.type !== MAPPING_RULE_TYPE_OBJECT &&
-                    (id === true || id === this.props.id)
-                ) {
-                    this.setState({expanded});
-                }
-            }
-        );
-        this.subscribe(
-            hierarchicalMappingChannel.subject('ruleView.change'),
-            this.onOpenEdit
-        );
-        this.subscribe(
-            hierarchicalMappingChannel.subject('ruleView.close'),
-            this.onCloseEdit
-        );
-        this.subscribe(
-            hierarchicalMappingChannel.subject('ruleView.discardAll'),
-            this.discardAll
-        );
-        if (this.state.isPasted)
-            this.props.scrollIntoView()
-    },
-    onOpenEdit(obj) {
+        EventEmitter.on(MESSAGES.RULE_VIEW.TOGGLE, this.handleToggleRule);
+        EventEmitter.on(MESSAGES.RULE_VIEW.CHANGE, this.onOpenEdit);
+        EventEmitter.on(MESSAGES.RULE_VIEW.CLOSE, this.onCloseEdit);
+        EventEmitter.on(MESSAGES.RULE_VIEW.DISCARD_ALL, this.discardAll);
+        
+        if (this.state.isPasted) {
+            this.props.scrollIntoView();
+        }
+    }
+    
+    componentWillUnmount() {
+        EventEmitter.off(MESSAGES.RULE_VIEW.TOGGLE, this.handleToggleRule);
+        EventEmitter.off(MESSAGES.RULE_VIEW.CHANGE, this.onOpenEdit);
+        EventEmitter.off(MESSAGES.RULE_VIEW.CLOSE, this.onCloseEdit);
+        EventEmitter.off(MESSAGES.RULE_VIEW.DISCARD_ALL, this.discardAll);
+    }
+    
+    handleToggleRule = ({ expanded, id }) => {
+        // only trigger state / render change if necessary
+        if (
+            expanded !== this.state.expanded &&
+            this.props.type !== MAPPING_RULE_TYPE_OBJECT &&
+            (id === true || id === this.props.id)
+        ) {
+            this.setState({ expanded });
+        }
+    };
+    
+    onOpenEdit = (obj) => {
```

Comment:
Please use constructor binding instead of arrow functions for now. A couple of reasons:
* more performant
* reachable for testing (with jest)
* not flagged as "experimental"
* unified style (rest of components use constructor binding)
* if we ever switch to a common lint rule set like airbnb it is forbidden by it anyway ;)

Note: There are many ongoing discussions out there like https://github.com/facebook/react/issues/9851#issuecomment-519324487

Reply:
yes right!